src/librustc_mir/transform/mod.rs

```diff
@@ -179,6 +191,11 @@ pub fn run_passes(
     }
 
     body.phase = mir_phase;
+
+    if mir_phase == MirPhase::Optimized {
+        validate::Validator { when: format!("end of phase {:?}", mir_phase) }
```

Comment:
So we run it here even without debug assertions? Isn't that expensive?

Reply:
The pass does very little work, so it's unlikely to be an issue

Reply:
> The pass does very little work

That is something I want to change.^^ https://github.com/rust-lang/rust/pull/72796 does some non-trivial type folding.